egs2/chime7_task1/asr1/run.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -euo pipefail
+calc_float() { awk "BEGIN{ printf \"%.12f\n\", $* }"; }
+calc_int() { awk "BEGIN{ printf \"%.d\n\", $* }"; }
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+######################################################################################
+# CHiME-7 Task 1 SUB-TASK 1 baseline system script: GSS + ASR using oracle diarization
+######################################################################################
+
+stage=3
+stop_stage=100
+
+# NOTE, use absolute paths !
+chime7_root=${PWD}/chime7_task1
+chime5_root= # you can leave it empty if you have already generated CHiME-6 data
+chime6_root=/raid/users/popcornell/CHiME6/espnet/egs2/chime6/asr1/CHiME6 # will be created automatically from chime5
+# but if you have it already it will be skipped
+dipco_root=${PWD}/../../chime7/task1/datasets/dipco # this will be automatically downloaded
+mixer6_root=/raid/users/popcornell/mixer6/
+
+# DATAPREP CONFIG
+manifests_root=./data/lhotse # dir where to save lhotse manifests
+cmd_dprep=run.pl
+# note with run.pl your GPUs need to be in exclusive mode otherwise it fails
+# to go multi-gpu see https://groups.google.com/g/kaldi-help/c/4lih8UKHBoc
+dprep_stage=0
+gss_dump_root=./exp/gss
+ngpu=4  # set equal to the number of GPUs you have, used for GSS and ASR training
+train_min_segment_length=1 # discard sub one second examples, they are a lot in chime6
+train_max_segment_length=20  # also reduce if you get OOM, here A100 40GB
+
+# GSS CONFIG
+gss_max_batch_dur=360 # set accordingly to your GPU VRAM, here A100 40GB
+cmd_gss=run.pl # change to suit your needs e.g. slurm !
+gss_dsets="chime6_train,chime6_dev,dipco_dev,mixer6_dev" # no mixer6 train in baseline
+
+
+# ASR CONFIG
+# NOTE: if you get OOM reduce the batch size in asr_config YAML file
+asr_stage=0 # starts at 13 for inference only
+asr_dprep_stage=0
+bpe_nlsyms="" # in the baseline these are handled by the dataprep
+asr_config=conf/tuning/train_asr_transformer_wavlm_lr1e-4_specaugm_accum1_preenc128_warmup20k.yaml
+inference_config="conf/decode_asr_transformer.yaml"
+lm_config="conf/train_lm.yaml"
+use_lm=false
+use_word_lm=false
+word_vocab_size=65000
+nbpe=500
+
+
+# and not contribute much (but you may use all)
+asr_max_epochs=8
+# ESPNet does not scale parameters with num of GPUs by default, doing it
+# here for you
+
+. ./path.sh
+. ./cmd.sh
+. ./utils/parse_options.sh
+
+asr_batch_size=$(calc_int 128*$ngpu) # reduce 128 bsz if you get OOMs errors
```

Comment:
added automatic scaling with number of GPUs. 
I scale batch size, max lr and also the warmup

Reply:
very cool!